chain/fee_manager_test.go

```diff
@@ -0,0 +1,194 @@
+	rules.EXPECT().GetMaxBlockUnits().Return(Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000})
+
+	// Similar to vm.go L326
+	minUnitPrice := rules.GetMinUnitPrice()
+
+	var d Dimensions
+	feeManager := NewFeeManager(nil)
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		// Set unit prices for different dimensions
+		feeManager.SetUnitPrice(i, minUnitPrice[i])
+		d[i] = uint64(i)
+	}
+
+	// Check unit prices
+	unitPrices := feeManager.UnitPrices()
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		require.Equal(unitPrices[i], feeManager.UnitPrice(i))
+	}
+
+	// Similar to spam.go
+	maxUnits := rules.GetMaxBlockUnits()
+	_, err := MulSum(unitPrices, maxUnits)
+	require.NoError(err)
+}
+
+func TestConsume(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	feeManager := NewFeeManager(nil)
+	chunks := 100
+
+	// Mock VM
+	rules := NewMockRules(ctrl)
+
+	// Similar to genesis
+	rules.EXPECT().GetMaxBlockUnits().Return(Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000})
+	rules.EXPECT().GetBaseComputeUnits().Return(uint64(1))
+	rules.EXPECT().GetStorageKeyReadUnits().Return(uint64(5))
+	rules.EXPECT().GetStorageValueReadUnits().Return(uint64(2))
+	rules.EXPECT().GetStorageKeyAllocateUnits().Return(uint64(20))
+	rules.EXPECT().GetStorageValueAllocateUnits().Return(uint64(5))
+	rules.EXPECT().GetStorageKeyWriteUnits().Return(uint64(10))
+	rules.EXPECT().GetStorageValueWriteUnits().Return(uint64(3))
+
+	// Get usage for bandwith, compute, read, allocate, writes
+	bandwidthOp := math.NewUint64Operator(200)
+	bandwidthUnits, err := bandwidthOp.Value()
+	require.NoError(err)
+
+	computeUnitsOp := math.NewUint64Operator(rules.GetBaseComputeUnits())
+	computeUnitsOp.Add(50)
+	computeUnitsOp.Add(100)
+	computeUnits, err := computeUnitsOp.Value()
+	require.NoError(err)
+
+	readsOp := math.NewUint64Operator(0)
+	readsOp.Add(rules.GetStorageKeyReadUnits())
+	readsOp.MulAdd(uint64(chunks), rules.GetStorageValueReadUnits())
+	readUnits, err := readsOp.Value()
+	require.NoError(err)
+
+	allocatesOp := math.NewUint64Operator(0)
+	allocatesOp.Add(rules.GetStorageKeyAllocateUnits())
+	allocatesOp.MulAdd(uint64(chunks), rules.GetStorageValueAllocateUnits())
+	allocateUnits, err := allocatesOp.Value()
+	require.NoError(err)
+
+	writesOp := math.NewUint64Operator(0)
+	writesOp.Add(rules.GetStorageKeyWriteUnits())
+	writesOp.MulAdd(uint64(chunks), rules.GetStorageValueWriteUnits())
+	writeUnits, err := writesOp.Value()
+	require.NoError(err)
+
+	consumed := Dimensions{bandwidthUnits, computeUnits, readUnits, allocateUnits, writeUnits}
+
+	// Similar to processor.go
+	ok, dimension := feeManager.Consume(consumed, rules.GetMaxBlockUnits())
+	require.True(ok)
+	require.Equal(dimension, Dimension(0))
+
+	// The fee required 
+	_, err = feeManager.MaxFee(consumed)
+	require.NoError(err)
+
+	// Update last consumed
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		feeManager.SetLastConsumed(i, consumed[i])
```

Comment:
This is already done as part of `Consume`: https://github.com/ava-labs/hypersdk/blob/797d38ae84c8d7dca027a6b1d3e875e49218082e/chain/fee_manager.go#L138